src/Symfony/Component/Security/Core/User/LdapUserProvider.php

```diff
@@ -113,30 +118,31 @@ public function supportsClass($class)
      */
     protected function loadUser($username, Entry $entry)
     {
-        $password = $this->getPassword($entry);
+        $password = $this->getAttributeValue($entry, $this->passwordAttribute);
 
         return new User($username, $password, $this->defaultRoles);
     }
 
     /**
-     * Fetches the password from an LDAP entry.
+     * Fetches a required unique attribute value from an LDAP entry.
      *
      * @param null|Entry $entry
+     * @param string     $attribute
      */
-    private function getPassword(Entry $entry)
+    private function getAttributeValue(Entry $entry, $attribute)
     {
-        if (null === $this->passwordAttribute) {
+        if (null === $attribute) {
```

Comment:
@quentinus95 By the way, you'll notice that the default value for `passwordAttribute` is actually null, which means that by default, the user provider expects not to have any password stored in the Entry, and will return a user with a null password.